### PR TITLE
COCOS-77 - Stream agent logs and events over gRPC

### LINKED ISSUE #77: Stream agent log and events through manager after connecting

### PR DESCRIPTION
**Pull request by SammyOina**
Refactor event handling and logging

Reworked event and log processing to use channels instead of direct HTTP calls. Removed obsolete events package and consolidated event structures, leading to cleaner and more maintainable code. Updated agent events to use channels, enhanced error handling in log forwarding, and simplified manager `New` function signature to accept an event channel directly.

- Removed `events` and `agentevents` packages to reduce complexity.
- Replaced direct event server communication with internal channel usage.
- Introduced `AgentEvent` struct in events.go for standardized event objects.
- Adapted `managerService` to dispatch events and logs through channels.
- Streamlined manager construction by removing the now-unnecessary event service and host IP parameters.

This change results in a more robust and easier to extend event and log management system within the agent-manager interaction.

resolves #77

### CODE REVIEW

File: cmd/manager/main.go [hunk 1]
```diff
@@ -15,44 +15,34 @@ import (
...
-	envPrefixGRPC      = "MANAGER_GRPC_"
-	envPrefixAgentGRPC = "AGENT_GRPC_"
-	envPrefixQemu      = "MANAGER_QEMU_"
-	defSvcGRPCPort     = "7001"
-	defSvcHTTPPort     = "9021"
+	svcName        = "manager"
+	envPrefixGRPC  = "MANAGER_GRPC_"
+	envPrefixQemu  = "MANAGER_QEMU_"
+	defSvcGRPCPort = "7001"
+	defSvcHTTPPort = "9021"
```
Comment (WashingtonKK) [inline cmd/manager/main.go:35]: Are these being used?

**Review by WashingtonKK — CHANGES_REQUESTED** [review]

File: cmd/manager/main.go (hunk 1 above)
Comment (SammyOina) [reply cmd/manager/main.go:35]: Removed in #80

**Review by WashingtonKK — APPROVED** [review]
LGTM

**Review by drasko — APPROVED** [review]
LGTM

**Pull request merged**